public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -68,6 +93,11 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
   const isRecordingRuleType = type === RuleFormType.cloudRecording;
   const isCloudAlertRuleType = type === RuleFormType.cloudAlerting;
 
+  const dispatch_ = useDispatch();
+  useEffect(() => {
+    dispatch_(fetchAllPromBuildInfoAction());
```

Comment:
A bit confused about this – can you talk about why this is required? And can we use `featureDiscoveryApi` here instead (using RTK Query)?

Reply:
this is needed because if not we don't have all the data sources loaded in Redux, and now we need them so we can filter by cloud and not cloud...

Reply:
we need to update the Redux store with the whole list, as we are using `CloudRulesSourcePicker` , that loads the list of cloud data sources from Redux state